docs/source/reference-core.rst

```diff
@@ -494,14 +494,11 @@ but *will* still close the underlying socket before raising
 Cancellation API details
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
-The primitive operation for creating a new cancellation scope is:
+:func:`move_on_after` and all the other cancellation facilities provided
+by Trio are ultimately implemented in terms of :class:`CancelScope`
+objects.
 
-.. autofunction:: open_cancel_scope
-   :with: cancel_scope
-
-Cancel scope objects provide the following interface:
-
-.. interface:: The cancel scope interface
+.. autoclass:: trio.CancelScope
```

Comment:
Normally in trio, we put the detailed class/method/property docs into the code as docstrings, and then import them via `autoclass` and friends.

Historically, we haven't done that for `CancelScope`, because of this hacky `.. interface:: The cancel scope interface` thing. But now that it's a regular class, maybe we should move the docs into regular docstrings?

Reply:
I did so for `deadline`, `shield`, and `cancel()`. Since `cancelled_caught` and `cancel_called` are regular attributes, not properties, there's not a great place to document them; and using `Attributes:` in the class docstring seems to force those attributes to show up first in the Sphinx output, which isn't what we want here.

Reply:
Works for me